src/main/java/com/scottlogic/sorts/AuthorPostSorter.java

```diff
@@ -16,17 +15,34 @@
     public List<UserPost> sort(List<UserPost> inputList, SortOrder orderIn) {
 
         List<UserPost> listToSort = new ArrayList<>();
+        List<UserPost> listSorted;
+
         if (inputList == null) {
             return listToSort;
         }
         listToSort = NullPostChecker.nullPostCheck(inputList);
 
-        switch (orderIn) {
+        List<UserPost> nullsInList = listToSort.stream().filter(post -> post.getAuthor() == null).collect(Collectors.toList());
+        List<UserPost> nonNullAuthorList = listToSort.stream().filter(post -> post.getAuthor() != null).collect(Collectors.toList());
+
+        listSorted = nonNullAuthorList.stream()
+                .sorted(orderIn.isAscending() ? AuthorName : AuthorName.reversed())
+                .collect(Collectors.toList());
+
+        if (!nullsInList.isEmpty()) {
```

Comment:
This condition could be replaced by checking the size of the sorted list against against the original. That way, we could copmute the non-null list only if we need it.